GUI.py

```
@@ -0,0 +1,748 @@
+
+class FilterThread(QThread):
+    finished = pyqtSignal(str)
+    error = pyqtSignal(str)
+
+    def __init__(self, command):
+        super().__init__()
+        self.command = command
+
+    def run(self):
+        try:
+            result = subprocess.run(self.command, capture_output=True, text=True)
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)